src/decoder/mod.rs

```diff
@@ -923,6 +934,60 @@ impl<R: Read + Seek> Decoder<R> {
         Ok(())
     }
 
+    pub fn read_jpeg(&mut self) -> TiffResult<DecodingResult> {
+        let offsets = self.get_tag_u32_vec(Tag::StripOffsets)?;
+        let bytes = self.get_tag_u32_vec(Tag::StripByteCounts)?;
+
+        let jpeg_tables: Option<Vec<u8>> = match self.find_tag(Tag::JPEGTables) {
+            Ok(None) => None,
+            Ok(_) => Some(self.get_tag_u8_vec(Tag::JPEGTables)?),
+            Err(e) => return Err(e),
+        };
+
+        if offsets.len() == 0 {
+            return Err(TiffError::FormatError(TiffFormatError::RequiredTagEmpty(
+                Tag::StripOffsets,
+            )));
+        }
+        if offsets.len() != bytes.len() {
+            return Err(TiffError::FormatError(TiffFormatError::Format(
+                String::from("Size mismatch of StripOffsets and StripByteCounts"),
```

Comment:
This unecessary allocates a string. The usual strip based decoding uses `TiffFormatError::InconsistentSizesEncountered` for this, please do so as well.

Reply:
done!